### PR TITLE
Allow to manually add NPM dependencies (peerDependencies, devDependencies and dependencies) #633

### PR DESCRIPTION
**Pull request by jjoslet**
This also allow to add dev and peer dependencies from an emitter extension

### CODE REVIEW

**Comment by vojtechhabarta** [comment]
I would like to request few changes:
- I just improved `Settings.convertToMap()` method to better report errors (e95f0039f78c8b4610758332afcb537abe4c4f80). Use this method instead of your copy.
- There are few typos in `GenerateMojo`

File: typescript-generator-maven-plugin/src/main/java/cz/habarta/typescript/generator/maven/GenerateMojo.java [hunk 1]
```diff
@@ -780,6 +780,33 @@
      */
     @Parameter
     private String npmBuildScript;
+    
+    /**
+     * List of additional NPM <code>dependencies</code>.<br>
+     * Only applicable when {@link #generateNpmPackageJson} parameter is <code>true</code> and generating implementation file (.ts).<br>
+     * Each item it this list specifies dependencies with its version.<br>
```
Comment (vojtechhabarta) [inline typescript-generator-maven-plugin/src/main/java/cz/habarta/typescript/generator/maven/GenerateMojo.java:787]: should be "dependency" (3x)
``` diff
- Each item it this list specifies dependencies with its version.<br>
+ Each item it this list specifies dependency with its version.<br>
```

**Comment by vojtechhabarta** [comment]
Sorry, one more think: do you think you could add validation to `Settings` that these new parameters are only used when `generateNpmPackageJson` is set to `true`? (probably somewhere before they are filled from extensions)

**Comment by jjoslet** [comment]
Your review has been pushed.

**Comment by vojtechhabarta** [comment]
Released in [2.30.840](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v2.30.840).

**Pull request merged**